app/models/application_instance.rb

```diff
@@ -63,15 +63,14 @@ def self.by_client_and_deployment(client_id, deployment_id, iss, lms_url)
   end
 
   def lti_defaults
-    domain = self.domain || Rails.application.secrets.application_main_domain
     config = lti_config.dup
     if config.present?
-      config[:launch_url] ||= "https://#{domain}/lti_launches"
-      config[:secure_launch_url] ||= "https://#{domain}/lti_launches"
-      config[:domain] ||= domain
-      config[:export_url] ||= "https://#{domain}/api/ims_exports.json"
-      config[:import_url] ||= "https://#{domain}/api/ims_imports.json"
-      config[:icon] ||= "https://#{domain}/#{config[:icon]}"
+      config[:launch_url] ||= launch_url
+      config[:secure_launch_url] ||= "https://#{get_domain}/lti_launches"
```

Comment:
Why not make this `launch_url` as well? Or do you change it downstream?

Reply:
I changed that as well